policy-controller/k8s/index/src/inbound/index.rs

```diff
@@ -176,14 +183,14 @@ struct NsUpdate<K, T> {
 impl Index {
     pub fn shared(cluster_info: impl Into<Arc<ClusterInfo>>) -> SharedIndex {
         let cluster_info = cluster_info.into();
-        Arc::new(RwLock::new(Self {
+        SharedIndex(Arc::new(RwLock::new(Self {
```

Comment:
This is a nit, but from an API point of view, the SharedIndex type (and the ensuing `index.0.clone()` change) is a bit icky.

Also, this file is already pretty large and we're adding another 100+ lines of incidentally related code.

We can address both of these concerns by:

* Adding a new module, policy-controller/k8s/index/src/inbound/index/metrics.rs
* With a `pub fn collector(pub super::SharedIndex) -> Box<dyn Collector> { ... }`